# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
# Commit: 93367fdd03ec
# Full Hash: 93367fdd03ec4b97b3a182f821b01fd53063196c
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 2
# Description:
#   Bug 1654112 - Update threading invariant of AudioConduit::GetLocalSSRCs to match mRecvStreamConfig comment. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102279
# ==============================================================================

diff -r 9b2bc72201da -r 93367fdd03ec dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Fri Jan 29 15:37:11 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Fri Jan 29 15:38:10 2021 +0100
@@ -98,7 +98,7 @@
 }
 
 std::vector<uint32_t> WebrtcAudioConduit::GetLocalSSRCs() {
-  MutexAutoLock lock(mMutex);
+  MOZ_ASSERT(NS_IsMainThread());
   return std::vector<uint32_t>(1, mRecvStreamConfig.rtp.local_ssrc);
 }
 